ref/Microsoft.Build/netstandard/Microsoft.Build.cs

```diff
@@ -1423,23 +1427,25 @@ public partial class CacheResult
     }
     public enum CacheResultType
     {
+        None = 0,
         CacheHit = 1,
         CacheMiss = 2,
         CacheNotApplicable = 3,
-        None = 0,
     }
     public abstract partial class PluginLoggerBase
     {
         protected PluginLoggerBase(Microsoft.Build.Framework.LoggerVerbosity verbosity) { }
         public abstract bool HasLoggedErrors { get; protected set; }
         public abstract void LogError(string error);
-        public abstract void LogMessage(string message, System.Nullable<Microsoft.Build.Framework.MessageImportance> messageImportance=default(System.Nullable<Microsoft.Build.Framework.MessageImportance>));
+        public abstract void LogMessage(string message, Microsoft.Build.Framework.MessageImportance? messageImportance = default(Microsoft.Build.Framework.MessageImportance?));
         public abstract void LogWarning(string warning);
     }
     [System.Runtime.InteropServices.StructLayoutAttribute(System.Runtime.InteropServices.LayoutKind.Sequential)]
-    public partial struct PluginTargetResult
+    public readonly partial struct PluginTargetResult
     {
-        public PluginTargetResult(string targetName, System.Collections.Generic.IReadOnlyCollection<Microsoft.Build.Framework.ITaskItem2> taskItems, Microsoft.Build.Execution.BuildResultCode resultCode) { throw null;}
+        private readonly object _dummy;
+        private readonly int _dummyPrimitive;
```

Comment:
Any idea why it's now adding these dummy fields?

Reply:
Great question, I have no idea! @epananth any ideas why this might happen 🙂? I believe this source generator is owned by arcade.

Reply:
Discussed with @riarenas, he mentioned that this lives in arcade but we do not own it. @ericstj might know more about this.

Reply:
These are added here for compile-time compat. This is added to structs that have private fields that need to be initialized. 

https://github.com/dotnet/arcade/blob/ca7fab569267ed3bc73360882d652d119aae5653/src/Microsoft.Cci.Extensions/Writers/CSharp/CSharpWriter.cs#L159-L184

If we don't include these it is observable, and potentially dangerous, impact on project that consume them:

- Allows pointers to be created for structs with fields of a reference type.
- Prevents the compiler from catching cyclic struct layout problems.
- Breaks C# definite assignment checking.
- Allows structs using [FieldOffset] to verify when they should not.
- Prevents developers from correctly planning for interop scenarios.

Here is the issue that made us add this: https://github.com/dotnet/runtime/issues/16402